cirq/ops/common_gates_test.py

```diff
@@ -152,6 +153,22 @@ def test_x_matrix():
                        np.array([[1 - 1j, 1 + 1j], [1 + 1j, 1 - 1j]]) / 2)
 
 
+def test_h_matrix():
+    assert np.allclose(((cirq.H**0.5)**2.0).matrix(), (cirq.H).matrix())
```

Comment:
This is not testing the matrix, it is testing that 0.5*2 = 1. You want something like this:

```python
sqrt = (cirq.H**0.5).matrix()
m = np.dot(sqrt, sqrt)
assert np.allclose(sqrt, cirq.H.matrix())
```

I would recommend also verifying that the handedness of the matrix is correct by checking the entries against the square root computed elsewhere.

This one is from quirk, by entering "X+Z" into the custom rotation gate and using an angle of 90 (with phase correction of 45) and then copying the raw values out of the address bar:

`{{0.8535534+0.1464466i,√⅛-√⅛i},{√⅛-√⅛i,0.1464466+0.8535534i}}`

Note: 0.853... = (1+sqrt(0.5))/2 and 0.146... = (1-sqrt(0.5))/2

Here's wolfram alpha:

![image](https://user-images.githubusercontent.com/79941/45189665-01c83c00-b1ef-11e8-85a4-21ed2e367795.png)